sklearn/metrics/tests/test_ranking.py

```diff
@@ -1468,3 +1469,129 @@ def test_partial_roc_auc_score():
         assert_almost_equal(
             roc_auc_score(y_true, y_pred, max_fpr=max_fpr),
             _partial_roc_auc_score(y_true, y_pred, max_fpr))
+
+
+@pytest.mark.parametrize('y_true, k, normalize, sample_weight, true_score', [
+    (
+        np.array([
+            [0, 0, 0, 0],
+            [0, 0, 0, 0],
+            [0, 0, 0, 0],
+            [0, 0, 0, 0],
+        ]),
+        2,
+        True,
+        None,
+        1
+    ),
+    (
+        np.array([
+            [1, 0, 0, 0],
+            [0, 1, 0, 0],
+            [0, 0, 1, 0],
+            [0, 0, 0, 1],
+        ]),
+        2,
+        True,
+        None,
+        .25
+    ),
+    ([0, 1, 2, 3], 2, False, None, 1),
+    ([0, 1, 2, 3], 2, True, [3, 1, 1, 1], .5),
+    ([0, 1, 2, 3], 3, True, None, .5),
+    ([0, 1, 2, 3], 3, False, None, 2),
+    ([0, 1, 2, 3], 3, False, [1, 2, 1, 1], 3),
+])
+def test_top_k_accuracy_score(y_true, k, normalize, sample_weight, true_score):
+    y_score = np.array([
+        [0.4, 0.3, 0.1, 0.2],
+        [0.3, 0.2, 0.3, 0.2],
+        [0.3, 0.4, 0.1, 0.2],
+        [0.3, 0.4, 0.2, 0.1],
+    ])
```

Comment:
Is this a relevant output for a multilabel case?

Typically, A DecisionTreeClassifier will output a list of len n_labels where each entry is an array of shape (n_samples, 2)

Reply:
It depends of the classifier you are using, some output a list of `n_classes` arrays of shape `(n_samples, 2)` as you said and some others output an array of shape `(n_samples, n_classes)`. See @jnothman [gist](https://gist.github.com/jnothman/4807b1b0266613c20ba4d1f88d0f8cf5) for an exhaustive summary.

Since `accuracy_score` expect an array of shape (n_samples, n_classes) for the multilabel case I went for the same solution

Reply:
> Since accuracy_score expect an array of shape (n_samples, n_classes) for the multilabel case I went for the same solution

It's quite different since `accuracy_score` expects predicted labels (as returned by `predict`), not predicted scores.

> See @jnothman gist for an exhaustive summary

Thanks, I wasn't even aware of these discrepancies. If we're going to support multilabel then, we should support and test all cases.

Your call, but it might be easier to first merge the PR without multilabel support, and then submit another one.